### DISCUSSION TITLE
Question: How to reset the Perspective Viewer component?

### DISCUSSION THREAD

**Discussion by tobilg** [perspective-dev/perspective#2587] [Q&A]
Not a bug report, but a question: How can I reset the Perspective Viewer components configuration? I use is in my https://sql-workbench.com project to show resultsets from DuckDB WASM (via Arrow).

When I run different queries, I sometimes see some strange behavior (no scrolling possible, view not updating etc.). Once I click on reset in the component, everything works again. Unfortunately I failed to understand how this can be achieved programmatically...

Basically, I run the following code once a query succeeded:

```typescript
// Load Apache Arrow data as table
const table = await worker.table(buffer);

// Load table
await viewerRef.current.load(table);
```

The schema of the Arrow data in `buffer` can change with every query. How can I programmatically reset the table's view, so that the same thing happens internally as pressing the "Reset" button in the component itself=

**Comment by timkpaine** [comment]
The docs have some details, note that `reset` is not what you want if you want to delete the underlying data in the table and replace it, you probably want [`delete`](https://perspective.finos.org/docs/obj/perspective-viewer/#delete) + `reset`. @texodus will clarify if im giving wrong info 😄

**Comment by mhkeller** [comment]
I had a [similar question here](https://github.com/finos/perspective/discussions/2354) (with some details of my setup). I don't see the same kind of scrolling issues that you are but I also don't think my solution is completely correct.
